src/html5/parser/document.rs

```diff
@@ -494,6 +495,17 @@ impl Document {
             NodeData::Document(_) => {
                 _ = writeln!(f, "{}Document", buffer);
             }
+            NodeData::DocType(DocTypeData {
```

Comment:
I want to get rid of this... here we generate the nodes as strings, but we also do this when generating the trees during testing, which does a lot of the same work